src/System.CommandLine.Tests/Invocation/CommandHandlerTests.cs

```diff
@@ -507,5 +507,101 @@ public class OverridenVirtualTestCommandHandler : VirtualTestCommandHandler
             public override Task<int> InvokeAsync(InvocationContext context)
                 => Task.FromResult(41);
         }
+
+        [Fact]
+        public static void FromBindingContext_forwards_invocation_to_bound_handler_type()
+        {
+            var command = new RootCommand
+            {
+                Handler = CommandHandler.FromBindingContext<BindingContextResolvedCommandHandler>()
+            };
+            command.Handler.Should().NotBeOfType<BindingContextResolvedCommandHandler>();
+            var parser = new CommandLineBuilder(command)
+                .ConfigureBindingContext(context => context.AddService<BindingContextResolvedCommandHandler>())
+                .Build();
+
+            var console = new TestConsole();
+            parser.Invoke(Array.Empty<string>(), console);
+            console.Out.ToString().Should().Be(typeof(BindingContextResolvedCommandHandler).FullName);
+        }
+
+        [Fact]
+        public static void Subsequent_call_to_configure_overrides_service_registration()
+        {
+            BindingContextCommandHandlerAction action = (handler, Console) =>
+            {
+                handler.Should().BeOfType<BindingContextCommandHandler2>();
```

Comment:
If the handler is not called, then this assertion won't be called. It's safer to capture a local from the closure and assert on its state at the end of the test. This is also more readable as people generally expect the assertion at the end of the test.

Reply:
resolved in https://github.com/dotnet/command-line-api/pull/1358/commits/765d3a57f17a5eb4b2235e026d05a3ad3ce12609